backend/app/api/src/helpers/UitpasTokenRepository.ts

```diff
@@ -0,0 +1,102 @@
+            const url = 'https://account-test.uitid.be/realms/uitid/protocol/openid-connect/token';
+            const myHeaders = new Headers();
+            myHeaders.append('Content-Type', 'application/x-www-form-urlencoded');
+            const params = new URLSearchParams({
+                grant_type: 'client_credentials',
+                client_id: this.uitpasToken.clientId,
+                client_secret: this.uitpasToken.clientSecret,
+            });
+            const requestOptions: RequestInit = {
+                method: 'POST',
+                headers: myHeaders,
+                body: params.toString(),
+            };
+            const response = await fetch(url, requestOptions).catch((error) => {
+                // Handle network errors
+                throw new SimpleError({
+                    code: 'uitpas_unreachable',
+                    message: `Network error when fetching UiTPAS token for client id ${this.uitpasToken.clientId}: ${error.message}`,
+                    human: $t(`We konden UiTPAS niet bereiken. Probeer het later opnieuw.`),
+                });
+            });
+            if (!response.ok) {
+                // Handle non-200 responses
+                throw new SimpleError({
+                    code: 'unsuccessful_response_fetching_uitpas_token',
+                    message: `Could not successfully obtain an UiTPAS token for client id ${this.uitpasToken.clientId}: ${response.statusText}`,
+                    human: $t(`Er is een fout opgetreden bij het verbinden met UiTPAS. Probeer het later opnieuw.`),
+                });
+            }
+            const json = await response.json().catch((error) => {
+                // Handle JSON parsing errors
+                throw new SimpleError({
+                    code: 'invalid_json_response_fetching_uitpas_token',
+                    message: `Invalid JSON response when fetching UiTPAS token for client id ${this.uitpasToken.clientId}: ${error.message}`,
+                    human: $t(`Er is een fout opgetreden bij het communiceren met UiTPAS. Probeer het later opnieuw.`),
+                });
+            });
+            if (!json.access_token || !json.expires_in || typeof json.expires_in !== 'number' || json.expires_in <= 0) {
+                // Handle invalid response
+                throw new SimpleError({
+                    code: 'invalid_response_fetching_uitpas_token',
+                    message: `Invalid response when fetching UiTPAS token for client id ${this.uitpasToken.clientId}: ${JSON.stringify(json)}`,
+                    human: $t(`Er is een fout opgetreden bij de communicatie met UiTPAS. Probeer het later opnieuw.`),
+                });
+            }
+            this.accessToken = json.access_token;
+            this.expiresOn = new Date((Date.now() + json.expires_in * 1000) - 10000); // Set expiration 10 seconds earlier to be safe
+        }
+        return this.accessToken!;
```

Comment:
```suggestion
        return this.accessToken;
```
Normaal is het uitroepteken niet nodig en zou TypeScript zelf moeten zien dat die altijd defined is - anders is er mogelijk iets mis in de code